src/worker.rs

```diff
@@ -60,74 +60,73 @@ impl<'a> SidekiqWorker<'a> {
         SidekiqWorker {
             id: String::from_utf8_lossy(&identity).to_string(),
             server_id: server_id.into(),
-            pool,
+            redis,
             namespace,
             queues,
             handlers,
-            middlewares,
             tx,
             rx,
             processed: 0,
             failed: 0,
         }
     }
 
-    pub fn work(mut self) {
+    pub async fn work(mut self) {
         info!("worker '{}' start working", self.with_server_id(&self.id));
-        // main loop is here
-        let rx = self.rx.clone();
-        let clock = tick(Duration::from_secs(1));
+        let mut last_sync = Instant::now();
         loop {
-            select! {
```

Comment:
I decided not to use tokio's `select!` here since it cancels the future of any branch that wasn't executed. For the Redis `brpop` that could mean the future is canceled after Redis has sent us data over the network but before it's been received (meaning that job would be lost). I did have a `select!` implementation in this PR originally, but I removed it in favor of a simple loop because it's much more understandable this way.